lib/ansible/utils/vars.py

```diff
@@ -167,31 +167,50 @@ def load_options_vars(version):
 
 def isidentifier(ident):
     """
-    Determines, if string is valid Python identifier using the ast module.
+    Determines if string is valid Python identifier.
     Originally posted at: http://stackoverflow.com/a/29586366
     """
 
     if not isinstance(ident, string_types):
         return False
 
-    try:
-        root = ast.parse(ident)
-    except SyntaxError:
+    if not ident.isidentifier():
         return False
 
-    if not isinstance(root, ast.Module):
+    if keyword.iskeyword(ident):
         return False
 
-    if len(root.body) != 1:
+    return True
+
+
+def isidentifier_PY2(ident):
+    """
+    Determines if string is valid Python identifier.
+    Originally posted at: http://stackoverflow.com/a/29586366
+    """
+
+    if not isinstance(ident, string_types):
         return False
 
-    if not isinstance(root.body[0], ast.Expr):
+    if not ident:
         return False
 
-    if not isinstance(root.body[0].value, ast.Name):
+    # Python 2 does not consider True, False and None reserved keywords so add them manually
+    if keyword.iskeyword(ident) or ident in ('True', 'False', 'None'):
         return False
 
-    if root.body[0].value.id != ident:
+    first = '_' + string.lowercase + string.uppercase
+    if ident[0] not in first:
         return False
 
+    other = first + string.digits
+    for ch in ident[1:]:
```

Comment:
wouldn't \w regex be more efficient here?

Reply:
I suspect so. This is just copied straight from stackoverflow which already improves what we have now. I'll play with it a bit more.